### PR TITLE
HADOOP-16026:Replace incorrect use of system property user.name

### PR DESCRIPTION
**Pull request by dineshchitlangia**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 995 | trunk passed |
| +1 | compile | 942 | trunk passed |
| +1 | checkstyle | 58 | trunk passed |
| +1 | mvnsite | 78 | trunk passed |
| +1 | shadedclient | 813 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 103 | trunk passed |
| +1 | javadoc | 68 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 45 | the patch passed |
| +1 | compile | 904 | the patch passed |
| +1 | javac | 904 | the patch passed |
| +1 | checkstyle | 57 | the patch passed |
| +1 | mvnsite | 75 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 656 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 108 | the patch passed |
| +1 | javadoc | 67 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 497 | hadoop-common in the patch passed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 5600 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-603/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/603 |
| JIRA Issue | HADOOP-16026 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  |
| uname | Linux 350c30af41f0 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / f235a94 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-603/1/testReport/ |
| Max. process+thread count | 1589 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-603/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/SshFenceByTcpPort.java [hunk 1]
```diff
@@ -236,7 +237,13 @@ private int getSshConnectTimeout() {
     
     public Args(String arg) 
         throws BadFencingConfigurationException {
-      user = System.getProperty("user.name");
+      try {
+        user = UserGroupInformation.getCurrentUser().getShortUserName();
```
Comment (toddlipcon) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/SshFenceByTcpPort.java]: Not so sure about this part of the change. The ssh fencer is logging in to a unix user, so probably the unix username is more important than the kerberos username for the daemon.

Comment (dineshchitlangia) [reply hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/SshFenceByTcpPort.java]: @toddlipcon - Good catch. Now that you make this point, I too, am not convinced that it needs to be changed here. Let me request @jojochuang for his thoughts.
Thanks for reviewing.

Comment (dineshchitlangia) [reply hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/SshFenceByTcpPort.java]: @jojochuang  Could you share your thoughts pls? Thanks!

Comment (jojochuang) [reply hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/SshFenceByTcpPort.java]: Right. We shouldn't change this one.

Comment (dineshchitlangia) [reply hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ha/SshFenceByTcpPort.java]: @toddlipcon , @jojochuang thanks for your inputs and review. Review comments have been addressed. Changes are ready for final review/commit. Thanks!

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 29 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1028 | trunk passed |
| +1 | compile | 1027 | trunk passed |
| +1 | checkstyle | 48 | trunk passed |
| +1 | mvnsite | 71 | trunk passed |
| +1 | shadedclient | 744 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 108 | trunk passed |
| +1 | javadoc | 74 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 48 | the patch passed |
| +1 | compile | 952 | the patch passed |
| +1 | javac | 952 | the patch passed |
| +1 | checkstyle | 45 | the patch passed |
| +1 | mvnsite | 63 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 623 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 108 | the patch passed |
| +1 | javadoc | 60 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 518 | hadoop-common in the patch passed. |
| +1 | asflicense | 36 | The patch does not generate ASF License warnings. |
| | | 5617 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-603/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/603 |
| JIRA Issue | HADOOP-16026 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  |
| uname | Linux 86a3a0eefe0b 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / dc013f7 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-603/2/testReport/ |
| Max. process+thread count | 1471 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-603/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 8 | https://github.com/apache/hadoop/pull/603 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/603 |
| JIRA Issue | HADOOP-16026 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-603/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**